dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManager.kt

```diff
@@ -34,35 +34,65 @@ class SingleDataRequestManager(
      * @return the stored data request object
      */
     @Transactional
-    fun processSingleDataRequest(singleDataRequest: SingleDataRequest): List<StoredDataRequest> {
+    fun processSingleDataRequest(singleDataRequest: SingleDataRequest): SingleDataRequestResponse {
         utils.throwExceptionIfNotJwtAuth()
-        dataRequestLogger.logMessageForReceivingSingleDataRequest(singleDataRequest.companyIdentifier)
+
+        if (singleDataRequest.reportingPeriods.isEmpty()) {
```

Comment:
I believe we should also make sure the framework has been selected as well

Reply:
Yes, you are 100% right. However, this is already done implicitly by the definition of the framework field: Since it is of type DataTypeEnum, it cannot be removed completely (must not be null) and also we cannot not insert an empty string or some other stupid value as it is checked whether the input is one of the enum entries